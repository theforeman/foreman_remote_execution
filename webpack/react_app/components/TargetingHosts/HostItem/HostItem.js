import React from 'react';
import PropTypes from 'prop-types';
import HostStatus from '../HostStatus/HostStatus.js';
import HostActions from '../HostActions/HostActions.js';

const HostItem = (props) => {
  const {
    name, link, status, actions,
  } = props.host;

  const renderName = () => {
    if (link !== '' || link !== undefined) {
      return (<a href={link}>{name}</a>);
    }
    return (<a href="#" className="disabled">{name}</a>);
  };

  return (
    <tr>
      <td className="host_name">
        { renderName() }
      </td>
      <td className="host_status">
        <HostStatus status={status} />
      </td>
      <td className="host_actions">
        <HostActions actions={actions} />
      </td>
    </tr>

  );
};

HostItem.propTypes = {
  host: PropTypes.shape({
    name: PropTypes.string.isRequired,
    link: PropTypes.string.isRequired,
    status: PropTypes.string.isRequired,
    actions: PropTypes.array,
  }),
};

export default HostItem;
