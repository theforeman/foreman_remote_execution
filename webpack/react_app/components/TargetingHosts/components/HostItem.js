import React from 'react';
import PropTypes from 'prop-types';
import { ActionButtons } from 'foremanReact/components/common/ActionButtons/ActionButtons';
import HostStatus from './HostStatus';

const HostItem = ({ name, link, status, actions }) => {
  const hostLink = link ? (
    <a href={link}>{name}</a>
  ) : (
    <a href="#" className="disabled">
      {name}
    </a>
  );

  return (
    <tr id={`targeting-host-${name}`}>
      <td className="host_name">{hostLink}</td>
      <td className="host_status">
        <HostStatus status={status} />
      </td>
      <td className="host_actions">
        <ActionButtons buttons={[...actions]} />
      </td>
    </tr>
  );
};

HostItem.propTypes = {
  name: PropTypes.string.isRequired,
  link: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
  actions: PropTypes.array,
};

HostItem.defaultProps = {
  actions: [],
};

export default HostItem;
