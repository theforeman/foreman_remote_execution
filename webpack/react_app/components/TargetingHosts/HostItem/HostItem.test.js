import { testComponentSnapshotsWithFixtures } from '@theforeman/test';
import HostItem from './HostItem';

const fixtures = {
  'renders HostItem': {
    host: {
      name: 'Host1',
      link: '/host1',
      status: 'success',
      actions: [],
    },
  },

};
describe('HostItem', () =>
  testComponentSnapshotsWithFixtures(HostItem, fixtures));
