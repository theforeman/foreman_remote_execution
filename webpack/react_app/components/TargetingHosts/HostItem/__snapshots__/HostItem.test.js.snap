// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HostItem renders HostItem 1`] = `
<tr>
  <td
    className="host_name"
  >
    <a
      href="/host1"
    >
      Host1
    </a>
  </td>
  <td
    className="host_status"
  >
    <HostStatus
      status="success"
    />
  </td>
  <td
    className="host_actions"
  >
    <HostActions
      actions={Array []}
    />
  </td>
</tr>
`;
